### DISCUSSION TITLE
Feature request: temporary interaction limits for vacation

### DISCUSSION THREAD

**Discussion by wlandau** [community/community#84052] [Repositories]
I am writing to request features to support the wellness and mental health of developers who go on vacation.

I maintain repositories such as https://github.com/ropensci/targets which are popular in my community. Whenever I go on vacation, I worry about the large number of tasks that pile up in my absence, and I dread the hours it will take to catch up when I return. So I end up spending vacation time to catch up on issues and discussions, which disrupts my time off and erodes my work/life balance.

To prevent the backlog, I now have a workaround: I set [temporary interaction limits](https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-repository) to prevent new activity on issues, discussions, and pull requests. In addition, I [open an issue](https://github.com/ropensci/targets/issues/1007) to inform users what is happening and why. With this approach, I am finally able to unplug and rest. However, there are two major problems:

1. I cannot customize the duration of interaction limits. I am forced to choose preset values of 24 hours, 3 days, 1 week, 1 month, or 6 months. If I go on vacation for 10 days, for example, I have to set a 1-month limit and then manually re-enable interactions when I return.
2. Most users do not know about my "vacation mode". Despite the [issue here](https://github.com/ropensci/targets/issues/1007), most people [still try to comment and then get confused when they cannot post anything](https://github.com/ropensci/targets/discussions/1200#discussioncomment-7859262).

I would like to request:

1. Custom durations for interaction limits.
2. A conspicuous customizable banner at the top of the repository page to let users know why interactions are temporarily limited.

There is precedent for (2) in other contexts. For example, when the Comprehensive R Archive Network (CRAN) goes on vacation, they replace their webform at https://cran.r-project.org/submit.html with a clear and obvious announcement:

![Screenshot 2024-01-03 at 10 38 39 AM](https://github.com/community/community/assets/1580860/d440fbd2-af6d-4297-99e6-2822bd9fa2d6)

The ability to post such a notice for repositories on GitHub would greatly improve the mental well-being of developers while also respecting the needs of users.

**Comment by wlandau** [comment]
I have requested these features in the past, but only as comments on existing threads:

* https://github.com/orgs/community/discussions/17599#discussioncomment-6961474
* https://twitter.com/wmlandau/status/1674982194987646976
* https://twitter.com/wmlandau/status/1610434642364055553
* https://fosstodon.org/@landau/110636699990847948

@HenSquared, is there someone on the @GitHub engineering team with whom I can discuss this?

**Comment by natalierjackson** [comment]
Thank you for your feedback, and we genuinely appreciate your input. Our team has reviewed your request and we want to be transparent about our priorities.This particular request is not currently being actively worked on. This doesn’t imply that we’re not considering it in the future, but for now, it is not on our development roadmap.We encourage you to check our [public roadmap](https://github.com/github/roadmap/projects/1) to get insights into the projects we are currently focusing on for the upcoming year. Please note that smaller fixes may not always be listed explicitly but can often be addressed when related work is in progress.

As a valuable GitHub user, your feedback in this Discussion space, accompanied by appropriate tags and examples, plays a crucial role in shaping GitHub’s development.While we can’t guarantee immediate action on every piece of feedback, rest assured that we are attentive to your suggestions and always striving to enhance the GitHub experience.

Once again, thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is valuable and contributes to the continuous improvement of GitHub
